### PR TITLE
Feat: Mark nonlist items via settings #113

### PR DESCRIPTION
**Pull request by GeorgeGo**
Add a settings setting to allow for marking of nonlist items

By default, feature is off.

In reference to this issue: https://github.com/ebullient/obsidian-task-collector/issues/113

### CODE REVIEW

**Comment by ebullient** [comment]
Thank you! I haven't had time to pick this one up..
Can you have a try at some tests? ;) A simple addition here would be nice: 
https://github.com/ebullient/obsidian-task-collector/blob/main/test/markTasks.test.ts

File: src/taskcollector-TaskCollector.ts [hunk 1]
```diff
@@ -208,6 +208,12 @@ export class TaskCollector {
                 mark
             );
         }
+        if (lineText && this.settings.convertEmptyLines) {
+            return this.updateLineText(
+                `- [ ] ${lineText}`,
```
Comment (ebullient) [inline src/taskcollector-TaskCollector.ts:213]: Some tests specifically here around the behavior/preservation of leading whitespace would be good (i.e. an indented line should remain indented, even when taskified.. )

Comment (GeorgeGo) [reply src/taskcollector-TaskCollector.ts:213]: Not something I originally considered, good catch!

**Comment by GeorgeGo** [comment]
Got a little turned around navigating PRs but I incorporated the tests you mentioned and adjusted for existing indent using your established regex.

**Pull request merged**